cmd/snapd-generator/main.c

```diff
@@ -97,3 +90,80 @@ int main(int argc, char **argv)
 	fprintf(f, "WantedBy=local-fs.target\n");
 	return 0;
 }
+
+int ensure_fusesquashfs_inside_container(const char *normal_dir)
+{
+	// check if we are running inside a container
+	if (system("systemd-detect-virt --container --quiet") != 0) {
+		return 0;
+	}
+
+	char *fstype = NULL;
+	if (system("which squashfuse > /dev/null") == 0) {
+		fstype = "fuse.squashfuse";
+	} else if (system("which snapfuse > /dev/null") == 0) {
+		fstype = "fuse.snapfuse";
+	} else {
+		fprintf(stderr,
+			"cannot find squashfuse or snapfuse executable\n");
+		return 2;
+	}
+
+	DIR *units_dir SC_CLEANUP(sc_cleanup_closedir) = NULL;
+	units_dir = opendir("/etc/systemd/system");
+	if (units_dir == NULL) {
+		fprintf(stderr,
+			"cannot open /etc/systemd/system directory: %m\n");
+		return 2;
```

Comment:
/etc/systemd/system does not need to exist, and if it doesn't and thus is empty it means no mount units need fixing up. So why is this an error then?

Reply:
Done.